crates/storage/provider/src/providers/static_file/jar.rs

```diff
@@ -104,12 +106,10 @@ impl<N: NodePrimitives<BlockHeader: Value>> HeaderProvider for StaticFileJarProv
         &self,
         range: impl RangeBounds<BlockNumber>,
     ) -> ProviderResult<Vec<Self::Header>> {
-        let range = to_range(range);
-
         let mut cursor = self.cursor()?;
-        let mut headers = Vec::with_capacity((range.end - range.start) as usize);
+        let mut headers = Vec::with_capacity(range_size_hint(&range).unwrap_or(1024));
```

Comment:
this can still be abused in theory at least I believe:`0..=u64::MAX` for example

but should be okay because we don't use it like this